bids_prov/fsl/fsl_parser.py

```diff
@@ -138,75 +138,210 @@ def readlines(filename: str) -> Mapping[str, List[str]]:
-    >>> parameters = [2, 4, "-a"]
-    >>> get_entities(cmd_s, parameters)
-    ['input1', 'input2', 'input1']
+
+    >>> df = {
+            "name": "command",
+            "used": [0, "-a"],
+            "generatedBy": [-1, "-b"]
+        }
+    >>> cmd_s = ["command", "-a", "kwarg_0", "arg_0", "arg_1",  "-b", "kwarg_1"]
+    >>> inputs, outputs, parameters = get_entities(cmd_s[1:], df)
+    >>> (["kwarg_0", "arg_0"], ["kwarg_1", "arg_1"], [])
+
     """
-    entities = []
-    args_consumed_list = []
-    for u_arg in parameters:
-        if type(u_arg) == int:
-            if not cmd_s[u_arg].startswith("-") :
-                entities.append(cmd_s[u_arg])   # the "if" is useful for
-            # Entities that are optional but indicated in the description file
-            # Example : "/slicer rendered_thresh_zstat2 -A 750 zstat2.png" with "used": [1, 2]
-            # Sometimes, 2 is present. In the previous command, this is not the case
-                args_consumed_list.append(cmd_s[u_arg])
-        elif type(u_arg) == dict:
-            # Allows us to retrieve entities not directly attached to the parameter name
-            # Example : "/slicer rendered_thresh_zstat2 -A 750 zstat2.png" with "generatedBy":
-            # [{"name": "-A", "index": 2}]
-            if u_arg["name"] in cmd_s:
-                entities.extend([cmd_s[i + u_arg["index"]] for i, cmd_part in enumerate(cmd_s) if cmd_part == u_arg["name"]])
-                # The for loop allows to retrieve the entities of the parameters appearing several times
-                # Example : /slicer example_func2highres highres -s 2 -x 0.35 sla.png -x 0.45 slb.png -x 0.55 slc.png
-        else:  # type(u_arg) == str
-            if u_arg in cmd_s:
-                entities.append(cmd_s[cmd_s.index(u_arg) + 1])  # we add the entity located just after the parameter
-                if u_arg.startswith("-") or '>' in u_arg :
-                    args_consumed_list.append(cmd_s[cmd_s.index(u_arg) + 1])
-                    args_consumed_list.append(u_arg)
-            elif not u_arg.startswith("-"):  # case of slicing
-                u_arg_splitted = u_arg.split(":")
-                start = int(u_arg_splitted[0])
-                if u_arg_splitted[1] == "":
-                    stop = None
-                else:
-                    stop = int(u_arg_splitted[-1])
 
-                if re.search(r"(-f|-rf)", cmd_s[1]):# to skip -r or -rf option
-                     add_ent = cmd_s[slice(start+1, stop)]
-                else:
-                    add_ent = cmd_s[slice(start, stop)]
-                entities.extend(add_ent)
-                args_consumed_list.extend(add_ent)
+    parser = argparse.ArgumentParser(
+        add_help=False, conflict_handler='resolve')
+
+    inputs_kwarg = []
+    outputs_kwarg = []
+    parameters_value = []
+    parameters_no_value = []
+
+    if "used" in parameters:
+        parser, inputs_kwarg = _get_kwarg(parser, parameters["used"])
```

Comment:
parser.add_argument outside function `_get_kwarg`

Reply:
Fix in commit **answer hermann**